lib/mox/server.ex

```diff
@@ -120,18 +124,25 @@ defmodule Mox.Server do
 
   def handle_call(
         {:fetch_fun_to_dispatch, caller_pids, {mock, _, _} = key, source},
-        %{mode: :private, lazy_calls: lazy_calls} = state
+        %{mode: :private} = state
       ) do
-    state = maybe_revalidate_lazy_calls(lazy_calls, state)
+    owner_pid =
+      case NimbleOwnership.get_owner(state.os, caller_pids, mock) do
+        %{owner_pid: owner_pid} -> owner_pid
```

Comment:
Can the owner_pid be nil? If not, shouldn't we merge the conditional below into this function?